ci: FreeBSD CI is currently broken

There's an issue setting up LLVM on the FreeBSD CI

Seems to be related to
https://github.com/RPCS3/rpcs3/blob/9a7b8e41eb741dd074c1a81d1cb80406c1f86a0a/.ci/build-freebsd.sh#L20

https://cirrus-ci.com/task/5486618942898176?logs=main#L104
```
CMake Error at modules/CMakeLists.txt:221 (file):
  file RELATIVE_PATH must be passed a full path to the directory:
  libcxx_prefix/lib
```

No obvious offender since Cirrus CI was mostly down last month due to `Monthly compute limit exceeded`, probably one of the LLVM related updates.

--- a/.ci/build-freebsd.sh
+++ b/.ci/build-freebsd.sh
@@ -12,7 +12,9 @@ tar xf llvm*.tar.xz
 export CC=clang19 CXX=clang++19
 cmake -B libcxx_build -G Ninja -S llvm*/libcxx \
       -DLLVM_CCACHE_BUILD=ON \
+      -DLIBCXX_CXX_ABI=libcxxrt \
       -DLIBCXX_INCLUDE_BENCHMARKS=OFF \
+      -DPython3_EXECUTABLE:FILEPATH="$(pkg query -x %Fp python3 | grep -Fm1 bin/python)" \
       -DCMAKE_INSTALL_PREFIX:PATH="$PWD/libcxx_prefix"
 cmake --build libcxx_build
 cmake --install libcxx_build

--- a/.ci/build-freebsd.sh
+++ b/.ci/build-freebsd.sh
@@ -19,7 +19,7 @@ cmake -B libcxx_build -G Ninja -S llvm*/libcxx \
 cmake --build libcxx_build
 cmake --install libcxx_build
 export CXXFLAGS="$CXXFLAGS -nostdinc++ -isystem$PWD/libcxx_prefix/include/c++/v1"
-export LDFLAGS="$LDFLAGS -nostdlib++ -L$PWD/libcxx_prefix/lib -l:libc++.a -lcxxrt"
+#export LDFLAGS="$LDFLAGS -nostdlib++ -L$PWD/libcxx_prefix/lib -l:libc++.a -lcxxrt"
 
 CONFIGURE_ARGS="
 	-DWITH_LLVM=ON

--- a/.ci/build-freebsd.sh
+++ b/.ci/build-freebsd.sh
@@ -5,22 +5,6 @@
 # shellcheck disable=SC2046
 git submodule -q update --init --depth 1 $(awk '/path/ && !/llvm/ && !/opencv/ { print $3 }' .gitmodules)
 
-# Prefer newer Clang than in base system (see also .ci/install-freebsd.sh)
-# libc++ isn't in llvm* packages, so download manually
-fetch https://github.com/llvm/llvm-project/releases/download/llvmorg-19.1.7/llvm-project-19.1.7.src.tar.xz
-tar xf llvm*.tar.xz
-export CC=clang19 CXX=clang++19
-cmake -B libcxx_build -G Ninja -S llvm*/libcxx \
-      -DLLVM_CCACHE_BUILD=ON \
-      -DLIBCXX_CXX_ABI=libcxxrt \
-      -DLIBCXX_INCLUDE_BENCHMARKS=OFF \
-      -DPython3_EXECUTABLE:FILEPATH="$(pkg query -x %Fp python3 | grep -Fm1 bin/python)" \
-      -DCMAKE_INSTALL_PREFIX:PATH="$PWD/libcxx_prefix"
-cmake --build libcxx_build
-cmake --install libcxx_build
-export CXXFLAGS="$CXXFLAGS -nostdinc++ -isystem$PWD/libcxx_prefix/include/c++/v1"
-#export LDFLAGS="$LDFLAGS -nostdlib++ -L$PWD/libcxx_prefix/lib -l:libc++.a -lcxxrt"
-
 CONFIGURE_ARGS="
 	-DWITH_LLVM=ON
 	-DUSE_SDL=OFF
@@ -31,6 +15,9 @@ CONFIGURE_ARGS="
 	-DUSE_SYSTEM_LIBPNG=ON
 	-DUSE_SYSTEM_OPENCV=ON
 "
+
+# base Clang workaround (missing clang-scan-deps)
+CONFIGURE_ARGS="$CONFIGURE_ARGS -DCMAKE_CXX_SCAN_FOR_MODULES=OFF"
 
 # shellcheck disable=SC2086
 cmake -B build -G Ninja $CONFIGURE_ARGS

--- a/.ci/build-freebsd.sh
+++ b/.ci/build-freebsd.sh
@@ -13,7 +13,7 @@ export CC=clang19 CXX=clang++19
 cmake -B libcxx_build -G Ninja -S llvm*/libcxx \
       -DLLVM_CCACHE_BUILD=ON \
       -DLIBCXX_INCLUDE_BENCHMARKS=OFF \
-      -DCMAKE_INSTALL_PREFIX:PATH=libcxx_prefix
+      -DCMAKE_INSTALL_PREFIX:PATH="$PWD/libcxx_prefix"
 cmake --build libcxx_build
 cmake --install libcxx_build
 export CXXFLAGS="$CXXFLAGS -nostdinc++ -isystem$PWD/libcxx_prefix/include/c++/v1"

--- a/.ci/install-freebsd.sh
+++ b/.ci/install-freebsd.sh
@@ -8,7 +8,7 @@ sed -i '' 's/quarterly/latest/' /etc/pkg/FreeBSD.conf
 export ASSUME_ALWAYS_YES=true
 pkg info # debug
 
-# Prefer newer Clang than in base system (see also .ci/build-freebsd.sh)
+# WITH_LLVM
 pkg install llvm19
 
 # Mandatory dependencies (qt6-base is pulled via qt6-multimedia)

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -103,7 +103,7 @@ freebsd_task:
   matrix:
     - name: Cirrus FreeBSD
       freebsd_instance:
-        image_family: freebsd-13-3
+        image_family: freebsd-13-4
         cpu: 8
         memory: 8G
   env:

--- a/rpcs3/util/cpu_stats.cpp
+++ b/rpcs3/util/cpu_stats.cpp
@@ -32,6 +32,7 @@
 
 #if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 # include <sys/sysctl.h>
+# include <unistd.h>
 # if defined(__DragonFly__) || defined(__FreeBSD__)
 #  include <sys/user.h>
 # endif

